### PR TITLE
add indexes for ordering by published time

### PR DESCRIPTION
**Pull request by mixmix**
**Semver Δ**: `minor`

**Problem**: on index pages of many apps (Ticktack, Patchbay gatherings) I need messages ordered by published time. This is particularly important for initial sync / identity restore, where messages are received from people in all sorts of order and can lead to a surprisingly disorienting view of what's going on

**Solution** : adds two indexes which add ability to `$filter` on `msg.value.timestamp` (asserted publish time). This PR puts these new indices below existing indices so that partial matches will behaviour is preserved (e.g. a filter on `type` and `author` will default to the existing `aty`index not to the new `ata` index)

### CODE REVIEW

**Comment by mixmix** [comment]
## Benchmarks

Existing index sizes 
```bash
$ du -h  ~/.ssb/flume | grep query
36M     /home/mix/.ssb/flume/query
```

New index size:
```bash
56M     /home/mix/.ssb/flume/query
```

For reference check out the sizes of the other indexes:
```bash
$ du -h  ~/.ssb/flume
75M     /home/mix/.ssb/flume/search
8.9M    /home/mix/.ssb/flume/internalblogStats
26M     /home/mix/.ssb/flume/keys
6.6M    /home/mix/.ssb/flume/time
444K    /home/mix/.ssb/flume/private-ye-QM09iP
9.4M    /home/mix/.ssb/flume/ticktack
18M     /home/mix/.ssb/flume/feed
4.0M    /home/mix/.ssb/flume/links2
126M    /home/mix/.ssb/flume/links
82M     /home/mix/.ssb/flume/backlinks-ye-QM09iP
56M     /home/mix/.ssb/flume/query
22M     /home/mix/.ssb/flume/patchwork-roots
7.9M    /home/mix/.ssb/flume/clock
```

**Comment by mixmix** [comment]
hey @dominictarr need this to complete Ticktack work.

Nice side effect is that with a filter like this we automatically get a stream of blogs in order published : 
```js
    $filter: {
      value: {
        content: { type: 'blog' },
        timestamp: { $gt: 1 },
      }
    }
```

File: index.js [hunk 1]
```diff
@@ -46,9 +49,10 @@ exports.init = function  (ssb, config) {
       opts = {query: JSON.parse(opts)}
     else if(isString(opts.query))
       opts.query = JSON.parse(opts.query)
+    debugger
```
Comment (dominictarr) [inline index.js:52]: you left an oopsie

**Comment by dominictarr** [comment]
merged into 2.1.0 (after removing debugger statement)

**Comment by mixmix** [comment]
Thanks for the prompt merge @dominictarr 

Now here's a fun thing - I need to get @mmckegg and all other people using ssb-query to update ssb-query, because otherwise that change of version of the index is going to mean a re-index every time a person starts up a client with a different version of ssb-query :crying_cat_face:

**Comment by dominictarr** [comment]
I think it will probably be an easy sell since maybe he will want these changes too ;)

**Comment by cryptix** [comment]
Hmmm... Nice!

Note to self: I need to copy this to your talequery plugin.

**Comment by dominictarr** [comment]
@cryptix talenet has it's own query plugin? please let me know about stuff like this, it really helps to know how people are using the database you are building (so that I can work on the features people need!)

**Comment by cryptix** [comment]
@dominictarr yes. I made a copy of ssb-query ([talequery](https://github.com/talenet/ssb-talequery/blob/master/index.js)) to give us the freedom to index message structures independently. In hindsight we could have just used `type:about` for a lot of stuff but we were moving to fast in the beginning.

I'll try to deprecate when we get around to v2 but migration of existing content will be a dozy...

I also wanted to refactor the reduce logic that now sits in the view into a FlumeReduce style plugin but that also has to wait until I can allocate viable time chunks to talenet again...

**Comment by mixmix** [comment]
Hey with migration of content it might be helpful to see if a pattern we
wrote for ssb-poll-schema might help. Basically we've explored writing
different versions of schema which support normalisation of messages. Cult
show you to parse content into a standard format meaning you won't lose
content.

Anyway, let me know if you'd like an intro to what we did. (hit me up via
ssb though)

**Comment by evbogue** [comment]
This is awesome.

**Pull request merged**